File: backend/ŠišAppApi/Models/AdminLog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ŠišAppApi.Models;

public class AdminLog
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int AdminId { get; set; }

    [ForeignKey("AdminId")]
    public Admin Admin { get; set; } = null!;

    [Required]
    [MaxLength(50)]
    public string Action { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    public string EntityType { get; set; } = string.Empty;

    public int? EntityId { get; set; }

    public string? OldValues { get; set; }

    public string? NewValues { get; set; }

    [MaxLength(255)]
    public string? Notes { get; set; }

    [MaxLength(50)]
    public string? IpAddress { get; set; }

    [MaxLength(255)]
    public string? UserAgent { get; set; }

    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public bool IsDeleted { get; set; } = false;

    public DateTime? DeletedAt { get; set; }
} 

File: sisapp_frontend/lib/models/admin_log.dart
import 'package:sisapp_frontend/utils/api_datetime.dart';

class AdminLog {
  final int id;
  final int adminId;
  final String adminName;
  final String action;
  final String entityType;
  final int? entityId;
  final String? notes;
  final String? ipAddress;
  final String? userAgent;
  final DateTime createdAt;

  AdminLog({
    required this.id,
    required this.adminId,
    required this.adminName,
    required this.action,
    required this.entityType,
    this.entityId,
    this.notes,
    this.ipAddress,
    this.userAgent,
    required this.createdAt,
  });

  factory AdminLog.fromJson(Map<String, dynamic> json) {
    return AdminLog(
      id: json['id'] as int,
      adminId: json['adminId'] as int,
      adminName: (json['adminName'] ?? '').toString(),
      action: (json['action'] ?? '').toString(),
      entityType: (json['entityType'] ?? '').toString(),
      entityId: json['entityId'] as int?,
      notes: json['notes']?.toString(),
      ipAddress: json['ipAddress']?.toString(),
      userAgent: json['userAgent']?.toString(),
      createdAt: ApiDateTime.parse(json['createdAt']),
    );
  }
}
